### PR TITLE
fix: do not panic when partition was removed from catalog

### PR DESCRIPTION
**Pull request by crepererum**
Fixes https://github.com/influxdata/idpe/issues/17040 .

### CODE REVIEW

File: querier/src/ingester/mod.rs [hunk 1]
```diff
@@ -580,13 +580,40 @@ async fn execute(
     decoder.finalize().await
 }
 
+#[derive(Debug)]
+enum CurrentPartition {
```
Comment (alamb) [inline querier/src/ingester/mod.rs:584]: Could we please add some comments about what each of these variants means (like that `Skip` means the data was from a different shard, for example?)

File: querier/src/ingester/mod.rs [hunk 2]
```diff
@@ -731,9 +766,13 @@ impl IngesterStreamDecoder {
...
                     partition_sort_key,
                 );
-                self.current_partition = Some(partition);
+                self.current_partition = CurrentPartition::Some(partition);
             }
             DecodedPayload::Schema(schema) => {
+                if self.current_partition.is_skip() {
+                    return Ok(());
+                }
+
```
Comment (alamb) [inline querier/src/ingester/mod.rs:775]: I wonder if it is feasible to cover the tests in this module with tests?

**Review by alamb — APPROVED** [review]

File: querier/src/ingester/mod.rs (hunk 1 above)
Comment (crepererum) [reply querier/src/ingester/mod.rs:585]: done

File: querier/src/ingester/mod.rs (hunk 2 above)
Comment (crepererum) [reply querier/src/ingester/mod.rs:781]: done

**Review by NGA-TRAN — COMMENTED** [review]
Nice. This solved the mystery I asked myself why removing partitions causes the trouble for Querier.

We support removing `namespace` recently that will cascade to remove its  tables and continue cascading to columns, partitions, parquet_files. Hope it won't affect Querier as well

**Comment by crepererum** [comment]
> We support removing `namespace` recently that will cascade to remove its tables and continue cascading to columns, partitions, parquet_files. Hope it won't affect Querier as well

Don't know about the querier, we've never tested this. It will for sure forever-loop/freeze the compactor.

**Comment by NGA-TRAN** [comment]
> Don't know about the querier, we've never tested this. It will for sure forever-loop/freeze the compactor.

This is scary, a ticket and a fix?

**Pull request merged**